### PR TITLE
Add ZFP plugin

### PR DESCRIPTION
**Pull request by oscargm98**

### CODE REVIEW

**Comment by oscargm98** [comment]
In case that reviewers  go to Files changed remember not checking the content of directories zfp/include and zfp/src.

File: include/blosc2.h [hunk 1]
```diff
@@ -985,6 +985,8 @@ static const blosc2_cparams BLOSC2_CPARAMS_DEFAULTS = {
   (zero) in the fields of the struct is passed to a function.
  */
 typedef struct {
+  int32_t typesize;
+  //!< The type size (8).
```
Comment (FrancescAlted) [inline include/blosc2.h:989]: Why `typesize` cannot be read for the blosc2 header?  Probably we discussed this, but could you refresh my memories?

File: plugins/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
 add_subdirectory(codecs)
 add_subdirectory(filters)
 
+set(SOURCES ${SOURCES} PARENT_SCOPE)
```
Comment (FrancescAlted) [inline plugins/CMakeLists.txt:4]: It is a good practice to always add an empty line at the end of a file.

File: plugins/codecs/codecs-registry.c [hunk 3]
```diff
@@ -18,4 +19,31 @@ void register_codecs() {
...
+
+    blosc2_codec zfp_rate;
+    zfp_rate.compcode = BLOSC_CODEC_ZFP_FIXED_RATE;
+    zfp_rate.compver = 1;
+    zfp_rate.complib = BLOSC_CODEC_ZFP_FIXED_RATE;
+    zfp_rate.encoder = blosc2_zfp_rate_compress;
+    zfp_rate.decoder = blosc2_zfp_rate_decompress;
+    zfp_rate.compname = "zfp_rate";
+    register_codec_private(&zfp_rate);
 }
```
Comment (FrancescAlted) [inline plugins/codecs/codecs-registry.c:49]: Ditto

**Review by FrancescAlted — COMMENTED** [review]
In general this looks good, except for the fact that `typesize` is now a part of the `blosc2_dparams` struct type.  Let's clarify this before going further.

File: include/blosc2.h (hunk 1 above)
Comment (oscargm98) [reply include/blosc2.h]: Changed in d62e4d8

File: plugins/codecs/codecs-registry.c (hunk 3 above)
Comment (oscargm98) [reply plugins/codecs/codecs-registry.c]: solved in d62e4d8

File: plugins/CMakeLists.txt (hunk 2 above)
Comment (oscargm98) [reply plugins/CMakeLists.txt]: solved in d62e4d8

**Review by FrancescAlted — APPROVED** [review]
Great job!  I detected some small glitches, but I prefer to merge this first and the do the fixes in a small commit.

**Pull request merged**